shiny/Data-Ingest/helper.R

```diff
@@ -0,0 +1,29 @@
+# This helper file stores all the relevant functions to be called in the shiny app
+#' Auto-create directory name for local file upload
+#'
+#' @param format_name Format name in type: character string
+#' @param site_id Integer, scientific notation, or character string that will be converted to an integer and will be the trailing identifier for the firectory name
+#'
+#' @return Self-generated directory name to store files uploaded via local upload. 
+#' @export
+#' @author Liam Burke (liam.burke24@gmail.com)
+#'
+#' @examples
+#' auto.name.directory(format_name = "LTER-hf-103", site_id = 1000004955)
+auto.name.directory <- function(format_name, site_id){
+  
+  # Split the format_name by the first non-alphanumeric character
+  splits <- base::strsplit("LTER-hf-103", "[^a-zA-Z0-9 :]")
+  
+  # Grab first split
+  basename <- splits[[1]][1]
+  
+  ## Convert site_id into shorter format
+  long_id <- as.numeric(site_id) # convert scientific notation to numeric
+  new_id <- base::sub('(?<=.{1}).+0', "-", long_id, perl = TRUE) # Replace all 0's with "-"
```

Comment:
This is not how the shorter format is generated. The break needs to specifically be at 1 billion. This version would incorrectly change machine 10 file 3 to 1-3

Reply:
Got it. Are there any example functions in the repo for how you would like me to do this?

Reply:
`paste0(new.site$id %/% 1e+09, "-", new.site$id %% 1e+09)`

Reply:
Thank you!!